smack-experimental/src/main/java/org/jivesoftware/smackx/hints/element/NoPermanentStoreHint.java

```diff
@@ -56,11 +56,7 @@ public static boolean hasHint(Message message) {
     }
 
     public static void set(Message message) {
-        if (StoreHint.hasHint(message)) {
-            // No need to set the no-permanent-store hint when a no-store hint is already set.
-            return;
-        }
-        setExplicitly(message);
+        message.overrideExtension(INSTANCE);
     }
 
     public static void setExplicitly(Message message) {
```

Comment:
I'm not sure if this method is still needed now, but I kept it in for now in order to not break the API.

Reply:
Help me here: Why do you think that it is no longer needed?

Reply:
Does it make sense to have multiple NoPermanentStoreHints in a message?
Users can still use `set` in order to set a hint on the message.

Reply:
I just noticed, after reading the comment within the function, that it checks for the store hint but this is the NoPermanentStore class. Hence I believe you should re-add the `if (StoreHint.hasHint(message)) return`.

Reply:
Readded the check